### DISCUSSION TITLE
Alternative Package management

### DISCUSSION THREAD

**Discussion by BCRARL** [fonsp/Pluto.jl#1357] [Questions about using Pluto] [answered]
I use DrWatson.j for project and package management and I use Pluto.jl to interactively develop scripts that will be integrated into the overall project. 

When I start Pluto it does not recognize the dependency for using statements following the activation of the new environment. How can I either set the environment ahead of time or force the dependency of using statements on the execution of the activation code?

**Comment by lungben** [comment, marked as answer]
You should be able to use shared environments with this pattern: https://github.com/fonsp/Pluto.jl/wiki/%F0%9F%8E%81-Package-management#pattern-the-shared-environment

**Reply by BCRARL** [reply]
Excellent that seems to have worked. Thank you!

**Reply by BCRARL** [reply]
Just to clarify, does this require that all packages to be used have to be in the first cell? That would be unfortunate as adding a package would require re-evaluation of the first cell.

**Reply by lungben** [reply]
You could split up the cells, but `Pkg.add("MyPkg")` und `using MyPkg` must be in the same cell for each package.
The cell `using Pkg; Pkg.activate(mktempdir())` could also be in a separate cell.

**Reply by BCRARL** [reply]
So on Pluto 0.15.1 I can't separate the using statements and still have them executed in the correct order, i.e., after the activate statement. I'll have to test whether prepending `using Pkg;` to using statements in the activated environment will reintroduce the order.
